spec/controllers/api/v1/accounts/bulk_actions_controller_spec.rb

```diff
@@ -0,0 +1,151 @@
+require 'rails_helper'
+
+RSpec.describe 'Api::V1::Accounts::BulkActionsController', type: :request do
+  include ActiveJob::TestHelper
+  let(:account) { create(:account) }
+  let(:agent_1) { create(:user, account: account, role: :agent) }
+  let(:agent_2) { create(:user, account: account, role: :agent) }
+
+  before do
+    Rails.application.config.active_job.queue_adapter = :inline
+
+    create(:conversation, account_id: account.id, status: :open)
+    create(:conversation, account_id: account.id, status: :open)
+    create(:conversation, account_id: account.id, status: :open)
+    create(:conversation, account_id: account.id, status: :open)
+  end
+
+  describe 'POST /api/v1/accounts/{account.id}/bulk_action' do
+    context 'when it is an unauthenticated user' do
+      let(:agent) { create(:user) }
+
+      it 'returns unauthorized' do
+        post "/api/v1/accounts/#{account.id}/bulk_actions",
+             headers: agent.create_new_auth_token,
+             params: { type: 'Conversation', fields: { status: 'open' }, ids: [1, 2, 3] }
+
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context 'when it is an authenticated user' do
+      let(:agent) { create(:user, account: account, role: :agent) }
+
+      it 'Bulk update conversation status' do
+        params = { type: 'Conversation', fields: { status: 'snoozed' }, ids: Conversation.first(3).pluck(:display_id) }
+
+        expect(Conversation.first.status).to eq('open')
+        expect(Conversation.last.status).to eq('open')
+        expect(Conversation.first.assignee_id).to eq(nil)
+
+        post "/api/v1/accounts/#{account.id}/bulk_actions",
+             headers: agent.create_new_auth_token,
+             params: { type: 'Conversation', fields: { status: 'snoozed' }, ids: %w[1 2 3] }
+
+        expect(response).to have_http_status(:success)
+
+        perform_enqueued_jobs do
```

Comment:
Since we are calling `perform` directly here, it beats the purpose of the `perform_enqueued_jobs` jobs block. It doesn't test the condition of whether the HTTP request was responsible for triggering the Bulkactions job.  

We should do either of the following 
1.  wrap the post request inside `perform_enqueued_jobs`
```
perform_enqueued_jobs do
  post "/api/v1/accounts/#{account.id}/bulk_actions",
             headers: agent.create_new_auth_token,
             params: { type: 'Conversation', fields: { status: 'snoozed' }, ids: %w[1 2 3] }
end

# this ensures that if code is correct bulkactionsjob will be triggered internally on the request
# then do expectations that the conversation objects are updated

```
2. rewrite the controller test to check whether the job is enqueued correctly and have a separate test for the bulk actions job  (  feels more right ) 


```
# psuedo code

# bulk actions controller spec 
bulk_actions_job = double
allow(BulkActionsJob).to receive(:new).and_return(bulk_actions_job)
allow(bulk_actions_job).to receive(:perform)

 post "/api/v1/accounts/#{account.id}/bulk_actions",
             headers: agent.create_new_auth_token,
             params: { type: 'Conversation', fields: { status: 'snoozed' }, ids: %w[1 2 3] }

expect(bulk_actions_job).to have_received(:perform).with(the parameters)
## tests that HTTP request enqueued correct job 


## Create a separate test file for bulk actions job where we test the various conditions


```